src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (true) {
+      case
+        obj.type === 'addProperties':
```

Comment:
You can write here `obj.type`, and in the following conditions it will not be necessary to specify it repeatedly.
And, can you rename obj?
If we iterate through an array of `actions` then the element of this array would be called `action`, right?

```suggestion
    switch (obj.type) {
      case 'addProperties':
```

Reply:
very well) done!